modules/gdscript/gdscript_parser.cpp

```diff
@@ -1895,6 +1888,7 @@ GDScriptParser::PatternNode *GDScriptParser::parse_match_pattern(PatternNode *p_
 			ExpressionNode *expression = parse_expression(false);
 			if (expression == nullptr) {
 				push_error(R"(Expected expression for match pattern.)");
+				return nullptr;
 			} else {
 				pattern->pattern_type = PatternNode::PT_EXPRESSION;
 				pattern->expression = expression;
```

Comment:
Since you removed the `PT_LITERAL` case, there should be a check if this expression is a literal to use that pattern type instead.

Reply:
Done